test.lua

```diff
@@ -781,6 +781,62 @@ do
     end
 end
 
+---- REPL tests ----
+print("Running tests for REPL completion...")
+local wrapRepl = function()
+    local replComplete
+    local replSend = coroutine.wrap(function()
+        local output = {}
+        fennel.repl({
+            readChunk = function()
+                local chunk = coroutine.yield(output)
+                output = {}
+                return chunk and chunk .. '\n' or nil
+            end,
+            onValues = function(xs)
+                table.insert(output, xs)
+            end,
+            registerCompleter = function(completer)
+                replComplete = function(input)
+                    local res = completer(input)
+                    return res
+                end
+            end,
+            pp = function(x) return x end,
+        })
+    end) replSend()
+    return replSend, replComplete
+end
+
+-- Skip REPL tests in non-JIT Lua 5.1 only to avoid engine coroutine bug
+if _VERSION ~= 'Lua 5.1' or type(jit) == 'table' then
+    local send, comp = wrapRepl()
+
+    send('(local [foo foo-bar* moe-larry] [1 2 {:*curly* "Why soitenly!"}])')
+    send('(local [!x-y !x_y] [1 2])')
+    for _, results in ipairs{
+        {comp'foo',  {'foo', 'foo-bar*'},
```

Comment:
Minor thing but the way the parens are omitted here bugs me; I think it'd be more readable with them added back.

Reply:
:+1:  I admittedly went back and forth on that one, shoulda listened to my inner voice.